spec/tasks/conditions/process_conditions_rake_spec.rb

```diff
@@ -83,24 +115,77 @@
         # should never create a SlackNotifier during this test
         expect(Slack::Notifier).not_to receive(:new)
 
-        expected_exception_log_entry = 'Exception: EOFError:  #<EOFError: EOFError>'
+        # precondition: log file should not exist
+        expect(File.exist?(filepath)).to be false
 
-        # precondition:
-        # log should not have the Exception
-        if File.exist?(filepath)
-          expect(File.read(filepath))
-              .not_to include expected_exception_log_entry
-        end
+        expect { subject.invoke }.not_to raise_error
 
-        expect { subject.invoke }.to raise_error EOFError
+        # post-conditions:
+        # 1) Log should have the processing Exception info
+        confirm_invalid_condition_exception(filepath, invalid_condition)
 
-        # post-condition:
-        # log should have the Exception
-        expect(File.read(filepath))
-            .to include expected_exception_log_entry
+        # 2) All valid conditions were processed successfully
+        confirm_valid_condition_processing(filepath, valid_conditions)
       end
 
     end
+
+    describe 'processing AND slack notification failures does not stop condition processing' do
+
+      it 'retries condition after Slack error, logs condition error and continues with remaining conditions' do
+
+        Condition.create!(invalid_condition)
+        output_loaded_condition_info
+
+        # Slack notification error that will be raised:
+        allow(SHFNotifySlack).to receive(:notification)
+                             .and_raise(Slack::Notifier::APIError)
+
+        # should never create a SlackNotifier during this test
+        expect(Slack::Notifier).not_to receive(:new)
+
+        # precondition: log file should not exist
+        expect(File.exist?(filepath)).to be false
+
+        expect { subject.invoke }.not_to raise_error
+
+        # post-conditions:
+        # 1) Log should have the processing Exception info
+        confirm_invalid_condition_exception(filepath, invalid_condition)
+
+        # 2) Log should have the Slack Notification Exception info
+        confirm_slack_notification_exception(filepath)
+
+        # 3) Log file indicates valid conditions were processed successfully
+        confirm_valid_condition_processing(filepath, valid_conditions)
+      end
+
+    end
+
+  end
+
+  def output_loaded_condition_info
+    puts "      #{Condition.count} conditions were loaded into the db: #{Condition.order(:class_name).map { |h_cond| h_cond[:class_name] }.join(', ')}"
+  end
+
+  def confirm_valid_condition_processing(filepath, conditions)
```

Comment:
I have mixed feelings about this. We should really be using a `shared_example` for this.  Or a `shared context`. 
Because then the tests read:  _it behaves like 'it does not have error messages in the log'_ which communicates exactly what the expected behavior should be.

But these methods are basically clear and readable enough.  Although I would prefer that they were named so that they communicated the expected behavior better.  "...valid condition processing" doesn't communicate that to me.  Maybe something like `confirm_no_error_entries_in_log` and `confirm_log_has_error_entry` for the method below.

But -- these are just suggestions.  Is fine with or without them.

Reply:
I did not want to create a shared context as I think the specifics of these expectations are very specific to this particular rake task.

Thanks for the other suggestion - I did rename some methods to be more reflective of what they are actually doing.